x-pack/plugins/observability/public/pages/alerts/render_cell_value.tsx

```diff
@@ -51,13 +56,36 @@ export const getRenderCellValue = ({
 }: {
   setFlyoutAlert: (data: TopAlert) => void;
 }) => {
-  return ({ columnId, data }: CellValueElementProps) => {
+  return ({ columnId, data, rowIndex, setCellProps }: CellValueElementProps) => {
+    const { isDarkMode, selectedAlertIndex, setSelectedAlertIndex } = useContext(TimelineContext);
     const { observabilityRuleTypeRegistry } = usePluginContext();
     const value = getMappedNonEcsValue({
       data,
       fieldName: columnId,
     })?.reduce((x) => x[0]);
 
+    const onExpandFn = useCallback(
+      (alert: TopAlert) => {
+        setFlyoutAlert?.(alert);
+        setSelectedAlertIndex?.(rowIndex);
+      },
+      [setSelectedAlertIndex, rowIndex]
+    );
+
+    useEffect(() => {
+      if (rowIndex === selectedAlertIndex) {
+        setCellProps({
```

Comment:
I think you'll want to do this on the leadingControlColumns as well, around https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/alerts/alerts_table_t_grid.tsx#L263